src/psyclone/tests/psyir/transformations/arrayrange2loop_trans_test.py

```diff
@@ -541,22 +447,245 @@ def test_character_validation(fortran_reader):
     with pytest.raises(TransformationError) as info:
         trans.validate(assign)
     assert (
-        "The ArrayRange2LoopTrans transformation doesn't allow "
-        "character arrays by default. This can be enabled by "
-        "passing the allow_string option to the transformation."
+        "ArrayRange2LoopTrans does not expand ranges on character arrays "
+        "by default (use the'allow_string' option to expand them), but found"
         in str(info.value))
 
     # Check we accept when we find character(LEN=x) syntax as this is an
     # UnsupportedFortranType
     # TODO #2441
     code = '''subroutine test()
-    character(LEN=100) :: a
-    character(LEN=100) :: b
-
-    a(1:94) = b(1:94)
+        character(LEN=100) :: a
+        character(LEN=100) :: b
+        a(1:94) = b(1:94)
```

Comment:
It was probably me who did this but this seems like a bad idea. Not for this PR but perhaps we should extend the frontend support slightly so that we get some partial-datatype information (e.g. that its intrinsic type is CHARACTER)? Alternatively, we could do what I've done in #2602 and actually look at the original declaration text?

Reply:
For now I created a new issue since the previous TODO was not exactly relevant.